packages/communication-react/review/beta/communication-react.api.md

```diff
@@ -1882,7 +1882,9 @@ export type CommonCallControlOptions = {
     cameraButton?: boolean | /* @conditional-compile-remove(PSTN-calls) */ {
         disabled: boolean;
     };
-    endCallButton?: boolean;
+    endCallButton?: boolean | /* @conditional-compile-remove(end-call-options) */ {
+        enableMenu: boolean;
```

Comment:
Q on feature impl, one customer wanted _only_ an end meeting for all button for the meeting host because they found other clients weren't leaving the call or didn't know they had to hit end call so they would run a long time.

Are we planning to only support an optional split button implementation?

Or do we have a plan to also make the entire end call button an end-call-for-all?

If so I would propose something like this for the API:

```ts
endCallButton?: boolean | {
  hangUpForEveryoneBehavior?: 'none' | 'primaryAction' | 'splitButtonAction'
}
```

Reply:
Or could go for something like this also that would better allow in the future switching the default action but still have hang up for self as a secondary split button action:
```ts
endCallButton?: boolean | {
  primaryAction: 'hangUp' | 'hangUpForEveryone',
  splitButtonActions: {
    hangUpForEveryone: boolean
  }
}
```

Reply:
According to current UX design - hang up for everyone is only for the action menu, maybe we could start with split button only then iterate it 

Override the primary button could introduce a strange scenario, for example, doctor wants to temporary leave then join back using another device, but it ended up with kicking everyone out of the meeting

Reply:
Going to schedule a weekly sync up soon, let's discuss this a bit more

 I like the first proposal, maybe we could go ahead with 
 
 hangUpForEveryoneBehavior?: 'none' |  'splitButtonAction'
 
 Then we can add | 'primaryAction' when whenever we want